### ISSUE TITLE
Inference with custom image shape 

### ISSUE THREAD

**Issue by sramirez** [rwightman/efficientdet-pytorch#40]
Hi, I tried to create predictions for a custom image shape other than that used for model training. I noticed that using img_size parameter in DetBenchPredict didn't output expected results (weird boxes).

Instead I got the desired results by relying on config parameter during model construction which is a bit clunky for me.

```
def load_net(checkpoint_path, image_size=512):
    config = get_efficientdet_config('tf_efficientdet_d6')
    net = EfficientDet(config, pretrained_backbone=False)
    config.num_classes = 1
    config.image_size=image_size
    net.class_net = HeadNet(config, num_outputs=config.num_classes, norm_kwargs=dict(eps=.001, momentum=.01))
    checkpoint = torch.load(checkpoint_path)
    net.load_state_dict(checkpoint['model_state_dict'])
    del checkpoint
    gc.collect()
    net = DetBenchPredict(net, config)
```

Seems img_size is not used in generate_predictions function, but it takes these info from the initialization parameters from initial config.

```
    **boxes = decode_box_outputs(box_outputs.float(), anchor_boxes, output_xyxy=True)**
    boxes = clip_boxes_xyxy(boxes, img_size / img_scale)  # clip before NMS better?
```

Any clue about how to improve this feature? Maybe I can help.

**Comment by rwightman** [comment]
you don't actually need to modify the config (model img size) for inference, you can just rely on scaling in image preprocessing (and setting img_size and img_scale appropriately for the prediction bench) and rescaling of the output boxes as is done for coco validation which has quite a variety of different input image sizes

if you have a full set of training data and you want to train a new model with a different aspect ratio you may want to consider changing the model image size, possibly to something non-square which I've yet to fully support.

**Comment by rwightman** [comment]
also note from the default image processing included, the model expects the image to be at the origin, upper left corner, and any any mismatch from the square input size to be dealt with by padding the lower and right sides. I pad with imagenet mean as it seems to improve results, the original TF automl code pads with black
